src/services/services.ts

```diff
@@ -1635,6 +1635,77 @@ module ts {
         sourceFile.scriptSnapshot = scriptSnapshot;
     }
 
+    export function transpile(input: string, compilerOptions?: CompilerOptions, fileName?: string, syntaxErrors?: Diagnostic[]): string {
+        let options = compilerOptions ? ts.clone(compilerOptions) : getDefaultCompilerOptions();
+
+        // Single file transformation is only guaranteed to be correct inside an external module.
+        // External modules have their own scope and cannot contribute to internal modules outside their
+        // scope. 
+        if (options.target !== ScriptTarget.ES6 && (!options.module || options.module === ModuleKind.None)) {

```

Comment:
We need to check that whatever we parse is an external module as well. else it should be an error.